build/apply-patches

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+#
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Run a command in a Docker container with devtoolset
+
+set -e
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+PATCH_DIR=${PATCH_DIR:-$(realpath "$SCRIPT_DIR/../patches/")}
+
+CUDF_DIR=${CUDF_DIR:-$(realpath "$SCRIPT_DIR/../thirdparty/cudf/")}
+
+pushd "$CUDF_DIR"
```

Comment:
nit: if we handle patches via `git apply` we can avoid switching working dirs by using `git -C $CUDF_DIR`

Reply:
I could do that with a patch too, but `git status` needs to be in the subdirectory/repo to work properly. `-C` does not work for `git status`.

Reply:
Maybe it is a bug in prior git versions 

```
~/gits/NVIDIA/spark-rapids-jni$ git --version
git version 2.34.1

~/gits/NVIDIA/spark-rapids-jni$ git -C thirdparty/cudf status -v --porcelain --untracked-files=no
 M print_env.sh
```

Reply:
I would rather stick with what I have tested for now.